### PR TITLE
refactor(radar_tracks_msgs_converter): rework parameters

### PR DESCRIPTION
**Pull request by PhoebeWu21**
## Description
Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371 for the `radar_tracks_msgs_converter` package.

- Remove the default value from the source code in order to ensure all parameter values are passed from the parameter files.
- Create the schema

## Tests performed
Package built and launch locally.
`colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release --packages-up-to radar_tracks_msgs_converter`

## Effects on system behavior
More reliable and faster parameter configuration file creation.

## Pre-review checklist for the PR author
The PR author **must** check the checkboxes below when creating the PR.
- [x] I've confirmed the [contribution guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/).
- [x] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).

## In-review checklist for the PR reviewers
The PR reviewers **must** check the checkboxes below before approval.
- [ ] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author
The PR author **must** check the checkboxes below before merging.
- [x] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: perception/radar_tracks_msgs_converter/schema/radar_tracks_msgs_converter.schema.json
Comment (kaspermeck-arm) [inline perception/radar_tracks_msgs_converter/schema/radar_tracks_msgs_converter.schema.json:18]: @PhoebeWu21 @mitsudome-r - I looked in `autoware.universe/perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp` to see if `new_frame_id` can be any value without success. Can this take any value?
If not, we could add an `enum` similar to as done in [refactor(joy_controller): rework parameters](https://github.com/autowarefoundation/autoware.universe/pull/3992/files#diff-a7768bad523a7c6d2571183d9240c85ca9829294291c048c8bec15df9b8ff5dfR13).

**Review by kaspermeck-arm — COMMENTED** [review]
Looks good, but need more input.

File: perception/radar_tracks_msgs_converter/schema/radar_tracks_msgs_converter.schema.json
Comment (PhoebeWu21) [reply perception/radar_tracks_msgs_converter/schema/radar_tracks_msgs_converter.schema.json:18]: Hi @kaspermeck-arm I think the value for `new_frame_id` can be assigned as any frame id name.

File: perception/radar_tracks_msgs_converter/config/config-radar_tracks_msgs_converter.param.yaml
Comment (scepter914) [inline perception/radar_tracks_msgs_converter/config/config-radar_tracks_msgs_converter.param.yaml:1]: [MUST]
Delete copyright as discussed at https://github.com/autowarefoundation/autoware.universe/pull/4695

File: perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp
Comment (scepter914) [inline perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp:1]: [MUST]
Delete copyright as discussed at https://github.com/autowarefoundation/autoware.universe/pull/4695

**Review by scepter914 — CHANGES_REQUESTED** [review]

File: perception/radar_tracks_msgs_converter/config/config-radar_tracks_msgs_converter.param.yaml
Comment (PhoebeWu21) [reply perception/radar_tracks_msgs_converter/config/config-radar_tracks_msgs_converter.param.yaml]: Done.

File: perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp
Comment (PhoebeWu21) [reply perception/radar_tracks_msgs_converter/src/radar_tracks_msgs_converter_node/radar_tracks_msgs_converter_node.cpp]: Done.

**Review by ambroise-arm — APPROVED** [review]

File: perception/radar_tracks_msgs_converter/config/config-radar_tracks_msgs_converter.param.yaml
Comment (kminoda) [inline perception/radar_tracks_msgs_converter/config/config-radar_tracks_msgs_converter.param.yaml:5]: Would you update the Readme as well so that it will automatically reads the schema file?

Comment (PhoebeWu21) [reply perception/radar_tracks_msgs_converter/config/config-radar_tracks_msgs_converter.param.yaml:5]: Done.

File: perception/radar_tracks_msgs_converter/README.md
Comment (scepter914) [inline perception/radar_tracks_msgs_converter/README.md:21]: [MUST]
This change is very inconvenience because we cannot see parameters description at github.
I think the parameter description should be written by markdown like [this PR](https://github.com/autowarefoundation/autoware.universe/pull/4871/files).

**Review by scepter914 — CHANGES_REQUESTED** [review]

File: perception/radar_tracks_msgs_converter/README.md
Comment (ambroise-arm) [reply perception/radar_tracks_msgs_converter/README.md:21]: @scepter914 This is an implementation of the goal described in https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/ros-nodes/parameters/#workflow. Could you consider using the rendered version of the documentation instead of the github pages? For example: https://autowarefoundation.github.io/autoware.universe/latest/perception/lidar_apollo_segmentation_tvm_nodes/#parameters

Comment (scepter914) [reply perception/radar_tracks_msgs_converter/README.md:21]: Why we use only the rendered version of documents?
I think it is convenience both github README and the rendered version.

In addition, as the reaction in PR of https://github.com/autowarefoundation/autoware.universe/pull/5241 is neglected for now, I cannot recognize the maintenance ability for using the json schema for parameter tables.
As code owner, I cannot accept the change that cause inconvenience for maintenance, and it is not allowed to delete the markdown parameter table unless there is enough maintenance ability.

**Comment by scepter914** [comment]
Because this PR has stale label and there are conflicts from many changes.
I close this PR once and ask to make new PR from now main branch.
Thank you for contributing.

**Comment by kaspermeck-arm** [comment]
@mitsudome-r @scepter914 

I don't see any issue with this PR. Would it be possible to re-open this issue and work on getting it merged? Thanks!

**Comment by scepter914** [comment]
@kaspermeck-arm 
Many PR are merged from the time when this PR made and there are conflicts.
Would you make new PR from now main branch?

**Comment by kaspermeck-arm** [comment]
> @kaspermeck-arm Many PR are merged from the time when this PR made and there are conflicts. Would you solving [this comment](https://github.com/autowarefoundation/autoware.universe/pull/4911#discussion_r1419847945) and make new PR from now main branch?

If I'm understanding correctly, there are two issues:
1. Since this PR was created, new code has been merged to the `radar_tracks_msgs_converter` which requires updating this PR
2. Unwillingness to adopt the new code contributions guidelines for parameters
    * E.g., single source of truth for documentation using the schema web documentation instead of README

@scepter914 - is this correct?

@mitsudome-r @ambroise-arm

**Comment by scepter914** [comment]
> 1. Since this PR was created, new code has been merged to the radar_tracks_msgs_converter which requires updating this PR

Yes.
I cannot merge this PR until the changes match the current main branch code.

> 2. Unwillingness to adopt the new code contributions guidelines for parameters

Because of many documentation update the parameter table is not enough to explain the parameters as you can see  [radar_object_clustering](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/radar_object_clustering).
So the replace from parameter documents of current version to the table made from json file like this PR cannot accept to merge.

As the parameter table change, I think the PR should not replace the parameter explain but only add tables.

Anyway, once this PR marked stale and there was a difference from the main branch as a result of being abandoned, so I think you should recreate the PR with a clean commit log as OSS develop.

**Comment by scepter914** [comment]
I cannot agree your suggestion.
The reason [radar_object_tracker](https://autowarefoundation.github.io/autoware.universe/main/perception/radar_object_tracker/#node-parameters) looks well for you is the few amount of explanation.
If we add the parameter many explanation like as [radar_object_clustering](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/radar_object_clustering), the table parameter is not suitable.

In summary, I cannot accept the change "replacing from raw README to parameter table made from a json file" and I can only accept the change "adding parameter table made from a json file with a short description" like below table.

| Name                 | Type   | Description                                                               | Default value |
| :------------------- | :----- | :------------------------------------------------------------------------ | :------------ |
| `angle_threshold`    | double | Angle threshold to judge whether radar detections come from one object    | 0.174         |
| `distance_threshold` | double | Distance threshold to judge whether radar detections come from one object | 4.0           |
| `velocity_threshold` | double | Velocity threshold to judge whether radar detections come from one object | 2.0           |
| `is_fixed_label`     | bool   | The flag to use fixed label                                               | false         |
| `fixed_label`        | string | The label of a clustered object is overwritten by this parameter          | "UNKNOWN"     |
| `is_fixed_size`      | bool   | The flag to use fixed size                                                | false         |
| `size_x`             | double | The x-axis size of a clustered object                                     | 4.0           |
| `size_y`             | double | The y-axis size of a clustered object                                     | 1.5           |
| `size_z`             | double | The z-axis size of a clustered object                                     | 1.5           |

**Pull request closed without merging**
